File: Ödev-2-/Ex_1_pthread1 1.c
#include <pthread.h>
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

#define NUM_THREADS	5

// Strings for the Hello World messages 
char *messages[NUM_THREADS];

// Thread işlevi
void *PrintHello(void *threadid) {
   int *myarg;
   sleep(1);                  // 1 saniye uyku
   myarg = (int *) threadid;  // İd'yi argümandan al
   printf("Thread %d: %s\n", *myarg, messages[*myarg]);
   pthread_exit(NULL);        // Thread'in sonlandırılması
}

int main(int argc, char *argv[]) {
  
  pthread_t threads[NUM_THREADS];    // Thread tanımlayıcıları 
  int i, rc, *taskid[NUM_THREADS];   // Her bir thread için id numaraları 

  // Her bir thread için mesajların atanması
  messages[0] = "English: Hello World!"; 
  messages[1] = "French: Bonjour, le monde!";
  messages[2] = "Spanish: Hola al mundo";
  messages[3] = "Klingon: Nuq neH!";
  messages[4] = "German: Guten Tag, Welt!"; 
   
  // Her bir thread için işlevin çağrılması ve thread'in oluşturulması
  for(i = 0; i < NUM_THREADS; i++) {
    // Thread'lere argüman olarak kullanılacak diziye bellek tahsisi
    taskid[i] = (int *) malloc(sizeof(int));
    *taskid[i] = i;
    // taskid[i] dizisindeki argümanı kullanarak thread oluşturma
    rc = pthread_create(&threads[i], NULL, PrintHello, (void *) taskid[i]);
    // Hata kontrolü
    if (rc) { 
      printf("ERROR; return code from pthread_create() is %d\n", rc);
      exit(-1);
    }
  }
  pthread_exit(0); // Ana thread'in sonlandırılması
} 

File: Ex_1_pthread1 1.cs
using System;
using System.Threading;
class Program
{
    // Toplam iş parçacığı sayısı
    const int thread_num = 5;

    // Farklı dillerdeki mesajlar
    static string[] messages = {
        "English: Hello World!",
        "French: Bonjour, le monde!",
        "Spanish: Hola al mundo",
        "Klingon: Nuq neH!",
        "German: Guten Tag, Welt!"
    };

    // Her bir iş parçacığı tarafından çalıştırılacak metot
    static void PrintHello(object id)
    {
        int threadId = (int)id;
        Thread.Sleep(500); // Her bir iş parçacığının 0,5 saniye uyumasını sağlar
        Console.WriteLine($"Thread {threadId}: {messages[threadId]}");
    }

    static void Main(string[] args)
    {
        // İş parçacıklarını depolamak için dizi
        Thread[] threads = new Thread[thread_num];

        // Her bir iş parçacığı oluşturulur ve başlatılır
        for (int i = 0; i < thread_num; i++)
        {
            int id = i;
            // Her bir iş parçacığına bir thread atanır ve başlatılır
            threads[i] = new Thread(() => PrintHello(id));
            threads[i].Start();
        }

        // Tüm iş parçacıklarının tamamlanmasını bekler
        foreach (Thread t in threads)
        {
            t.Join();
        }
    }
}